source/StarField.cpp

```diff
@@ -153,27 +166,43 @@ void StarField::Draw(const Point &pos, const Point &vel, double zoom) const
 	DrawList drawList;
 	drawList.Clear(0, zoom);
 	drawList.SetCenter(pos);
+
+	if(transparency > 0.)
+		transparency -= FADE_PER_FRAME;
+	else
+	{
+		transparency = 0.;
+	}
 	
 	// Any object within this range must be drawn. Some haze sprites may repeat
 	// more than once if the view covers a very large area.
 	Point size = Point(1., 1.) * haze.front().Radius();
 	Point topLeft = pos + (Screen::TopLeft() - size) / zoom;
 	Point bottomRight = pos + (Screen::BottomRight() + size) / zoom;
-	for(const Body &it : haze)
+	DrawHaze(haze, topLeft, bottomRight, transparency, drawList);
+	if(transparency > 0.)
+		DrawHaze(prevHaze, topLeft, bottomRight, 1 - transparency, drawList);
+
+	drawList.Draw();
+}
+
+
+void StarField::DrawHaze(const std::vector<Body> targetHaze, const Point targetTopLeft, const Point targetBottomRight, const double targetTransparency, DrawList& targetDrawList) const
```

Comment:
Note that without the `&` to declare the parameter is a reference, this still creates a copy of the input vectors. Also note that it isn't helpful to rename all parameters--doing so causes "churn"--just those parameters that are _class member variables_, like `haze`. The other parameters here were local variables created in the calling function: there's no chance the parameter can be renamed without also renaming the existing usages.

```suggestion
void StarField::DrawHaze(const std::vector<Body> &targetHaze, const Point &topLeft, const Point &bottomRight, double transparency, DrawList &drawList) const
```

You could actually go further, and make this function into a lambda or a function in the anonymous namespace, because it doesn't reference any class-specific information--only what is given to it.